refl1d/experiment.py

```diff
@@ -379,11 +388,19 @@ def __init__(self, sample=None, probe=None, name=None,
         self.dA = dA
         self.step_interfaces = step_interfaces
         self.interpolation = interpolation
-        num_slabs = len(probe.unique_L) if probe.unique_L is not None else 1
+        # TODO: proper 2D mesh resolution over L, T 
+        # (currently R is only calculated for first L anyway)
+        # num_slabs = len(probe.unique_L) if probe.unique_L is not None else 1
+        num_slabs = 1
         self._slabs = profile.Microslabs(num_slabs, dz=dz)
         self._probe_cache = material.ProbeCache(probe)
         self._cache = {}  # Cache calculated profiles/reflectivities
-        self._name = name
+        self.name = name if name is not None else probe.name
+        self.constraints = constraints
+        self.version = __version__ if version is None else version
+        if auto_tag:
+            tag_all(self.probe.parameters(), 'probe')
+            tag_all(self.sample.parameters(), 'sample')
```

Comment:
For simplicity I collapse the parameters into a flat list when I walk the model in bumps. Maybe this tagging could happen during flattening?

Reply:
What would be gained by doing the tagging during flattening?  You would have to keep track of the "auto_tag" setting (it would become part of the state of the class).

Reply:
If every application using bumps serialization needs tagging, then tagging is a bumps concern, not an application concern. [Not that I know what tagging does, but it doesn't seem like it would be specific to reflectometry.]